IvanCybercom/Hackademy-IV-frontend
# Important app upgrade: please read and test before approving

# This is a major update for angular and dependencies, it also includes some code restructuring. 

# Testing instructions:

  - `git pull app-upgrade`
  - `npm install`
  - `ng serve (and check everything)`
  - `ng test (and run all tests)`

# Changes

Updated angular and its libraries from 4.2.4 to 4.4.4
updated angular/cli from 1.3.2 to 1.4.5, and made it updatable.
updated angular/material from 2.0.0-beta.10 to 2.0.0-beta.12
updated zone.js from 0.8.14 to 0.8.18
added webpack as a local dev dependency

updated all references to material modules and components to use the new mat namespace in all modules and components

moved all material imports in admin module to the shared module
moved forms imports from admin module to the shared module
moved browser imports from admin module to the shared module
removed http import from admin module
removed material imports from all spec files as they are imported from the shared module
Fixed imports and declaration in the navbar spec file
